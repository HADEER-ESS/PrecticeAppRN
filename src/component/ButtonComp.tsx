import React from 'react'
import { StyleSheet, Text, TouchableOpacity } from 'react-native'

interface btnProps {
    btnText : String,
    btnPressFunction : ()=>void
}

const ButtonComp = ({btnText , btnPressFunction} : btnProps) => {
  return (
    <TouchableOpacity style={styles.btnStyle} onPress={btnPressFunction}>
        <Text style={styles.btnText}>{btnText}</Text>
    </TouchableOpacity>
  )
}

export default ButtonComp

const styles = StyleSheet.create({
    btnStyle : {
        backgroundColor : "#ff77ff",
        borderRadius : 17,
        marginHorizontal : "21%",
    },
    btnText:{
        fontSize : 24,
        padding : 11,
        color : "#ffffff",
        textAlign : 'center'
    }
})