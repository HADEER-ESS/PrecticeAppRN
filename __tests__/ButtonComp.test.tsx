import 'react-native';
import React from 'react';
import renderer, { ReactTestRendererJSON } from "react-test-renderer"
import {fireEvent, render} from "@testing-library/react-native"
import ButtonComp from '../src/component/ButtonComp';

describe("<ButtonComp/>" , ()=>{
    const mockedFun = jest.fn()
    const btn = renderer.create(<ButtonComp btnText={"Click"} btnPressFunction={mockedFun}/>)

    it("MAke sure the element render successfully" , ()=>{
        expect(btn).toMatchSnapshot()
    })

    it("The button fires" , ()=>{
        const {getByText} = render(
            <ButtonComp btnText={"Click"} btnPressFunction={mockedFun}/>
        )

        fireEvent.press(getByText("Click"))

        expect(mockedFun).toBeCalled()
    })
})