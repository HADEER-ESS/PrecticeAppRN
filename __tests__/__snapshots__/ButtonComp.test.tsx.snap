// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonComp/> MAke sure the element render successfully 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#ff77ff",
      "borderRadius": 17,
      "marginHorizontal": "21%",
      "opacity": 1,
    }
  }
>
  <Text
    style={
      {
        "color": "#ffffff",
        "fontSize": 24,
        "padding": 11,
        "textAlign": "center",
      }
    }
  >
    Click
  </Text>
</View>
`;
